resources/articles/s2s-auth.md

```diff
@@ -0,0 +1,121 @@
+[self service](/mobile/selfservice/) tool for the resource server.
+
+The scopes passed will help the receiving service authorize the call, and can be thought of a set of
+permissions or policies that the service will allow. Thus, the scopes present in the token help the
+called service determine if the request actions should be permitted. The exact scopes required is
+determined by the resource server and is outside the scope of this document.
+
+**Note:** currently the scopes that a client may request may only be managed by the administrators of
+the SPiD platform. In order to have scopes added to your client [contact support](mailto:support@spid.no).
+In the future, management of scopes will be possible via the [self service](/mobile/selfservice/) tool.
+
+## Send the token in the request
+
+Next, when making the request to the resource server, the caller simply needs to include the freshly
+acquired token in the Authorization header as a [bearer token](https://tools.ietf.org/html/rfc6750):
+
+```sh
+curl -H 'Authorization: Bearer _your_token_here' https://service.schibsted.com/messages/123
+```
+
+## Validate the token
+
+The resource server receives the token in `Authorization` header and must validate it in order to decide
+whether or not to allow the call. The first step in doing so is to [introspect the token](/token-introspection/)
+in order to get accessed to the claims it carries. Introspecting the token will provide a result similar
+to the following:
+
+```json
+{
+  "scope": "read:messages",
+  "client_id": "4e8463569caf7ca019000007",
+  "token_type": "Bearer",
+  "exp": 1487169930,
+  "iat": 1487166577,
+  "iss": "https://login.schibsted.com",
+  "aud": ["https://service.schibsted.com/messages"]
+}
+```
+
+One the resource server has introspected the token, it's important to validate the claims as follows:
+1. Has the token expired? This is determined by checking the `exp` claim of the token. When performing
+remote introspection this is handled automatically as the introspection request will return `false`
+as the value of the `active` property.
+1. Is the token issued by the expected issuer? If the resource server is using https://login.schibsted.com as
+its identity provider, it should only trust tokens with that URL in the `iss` claim.
+1. Does the token have the expected audience (determined by inspecting the `aud` claim, which is an array)?
+This is used to check if the token was intended for this service or not.
+1. Does the token have the expected scopes. The `scope` property contains a space-separated list of
+the scopes carried by the token. For each resource/action provided by the service, the implementers
+should decide on a set of scopes that are required to perform that action and ensure that the token
+carries those scopes.
+
+If all of the above checks pass, the resource server should let the call through. If the token has expired,
```

Comment:
Maybe include that missing or incorrect Authorization header should also result in `401 Unauthorized`.

Reply:
Fixed